syntax = "proto3";


option objc_class_prefix = "CD";

package contract_details;

service ContractDetailsService {
  rpc GetContractDetails(Request) returns (ContractDetails) {}
  rpc AddContractDetails(ContractDetails) returns(Empty){}
}

message Request {
    double conId = 1;
}

message Empty {}

message ContractDetails {
        int32 contractId = 1;
        string marketName = 2;
        float minTick = 3;
        string orderTypes = 4;
        string validExchanges = 5;
        int32 priceMagnifier = 6;
        int32 underConId = 7;
        string longName = 8;
        string contractMonth = 9;
        string industry = 10;
        string category = 13;
        string subcategory = 14;
        string timeZoneId = 15;
        string tradingHours = 16;
        string liquidHours = 17;
        string evRule = 18;
        int32 evMultiplier = 19;
        int32 mdSizeMultiplier = 20;
        int32 aggGroup = 21;
        string underSymbol = 22;
        string underSecType = 23;
        string marketRuleIds = 24;
//
//        TO COMPLETE LATER ONCE I KNOW WHAT IT IS
//
        string secIdList = 25;

        string realExpirationDate = 26;
        string lastTradeTime = 27;

//        # BOND values
        string cusip = 28;
        string ratings = 29;
        string descAppend = 30;
        string bondType = 31;
        string couponType = 32;
        bool callable = 33;
        bool putable = 34;
        int32 coupon = 35;
        bool convertible = 36;
        string maturity = 37;
        string issueDate = 38;
        string nextOptionDate = 39;
        string nextOptionType = 40;
        bool nextOptionPartial = 41;
        string notes = 42;
}